omnigibson/systems/macro_particle_system.py

```diff
@@ -155,95 +140,60 @@ def particle_name_prefix(cls):
 
     @classproperty
     def state_size(cls):
-        # We have n_particles (1), each particle pose (7*n), scale (3*n), and
-        # possibly template pose (7), and template scale (3)
-        state_size = 10 * cls.n_particles + 1
-        return state_size if cls.particle_object is None else state_size + 10
+        # In additon to super, we have:
+        # scale (3*n), and template pose (7), and template scale (3)
+        return super().state_size + 3 * cls.n_particles + 10
 
     @classmethod
     def _dump_state(cls):
-        return dict(
-            n_particles=cls.n_particles,
-            poses=[particle.get_local_pose() for particle in cls.particles.values()],
-            scales=[particle.scale for particle in cls.particles.values()],
-            template_pose=cls.particle_object.get_local_pose() if cls.particle_object is not None else None,
-            template_scale=cls.particle_object.scale if cls.particle_object is not None else None,
-        )
+        state = super()._dump_state()
+
+        state["scales"] = np.array([particle.scale for particle in cls.particles.values()])
+        state["template_pose"] = np.concatenate(cls._particle_object.get_local_pose())
+        state["template_scale"] = cls._particle_object.scale
```

Comment:
Sorry I know these are from before, but why do we need template_pose and template_scale again?

Reply:
delete, see Slack

Reply:
done